c10/util/ArrayRef.h

```diff
@@ -184,6 +184,9 @@ class ArrayRef final {
   /// @name Operator Overloads
   /// @{
   constexpr const T& operator[](size_t Index) const {
+    if (Index < 0) {
+      Index += Length;
+    }
```

Comment:
I'm... a little concerned about the performance implications here, as previously operator[] was unchecked and now it is checked. I suppose it is not too dangerous as the prevailing use of ArrayRef for sizes means that you aren't going to do very many accesses (so it's fast), but, if we are going to test for negative, we might as well ALWAYS do bounds checks (and then offer an unsafe fast index option.)